blog/posts/swiip/the-weight-of-js.md

```diff
@@ -0,0 +1,191 @@
+
+## The weight of things
+
+In order to being able to manage your app dependencies considering their weight, you have to know how much things weigh.
+
+There is popular plugins like [VS Code Import Cost](https://marketplace.visualstudio.com/items?itemName=wix.vscode-import-cost) or [Webpack Bundle Analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) to show the weight of each import you do in your code.
+
+![Import Code](./assets/weight-of-js/import-cost.png)
+
+![Bundle Analyzer](./assets/weight-of-js/webpack-bundle-analyzer.png)
+
+They are pretty fancy but beware because the problem is that worrying of weight in your code editor is too late. You have to consider the matter before this when you choose your stack.
+
+As explained before, there is different measures used commonly. Mainly before and after compression. To simplify, I will use the figures before compression (but after minification) for now on.
+
+### Common frameworks and library
+
+Without larger comments, let see the weight of the most popular libraries commonly used nowadays (sorry if there is not yours)
+
+- Lodash: 72kB
+- jQuery: 86kB
+- Backbone: 42kB (including Underscore but will need jQuery too)
+- AngularJS: 172kB
+- React: 117kB
+- Angular: 299kB (bundle of a Hello World app) with AoT, 663kB with JiT
+- Vue: 91kB
+- Preact: 8kB
+- Ember: 733kB (bundle of a Hello World app)
+- Leaflet: 138kB
+- Bootstrap: 152kB of CSS, 57kB of JS
+- Bulma: 186kB of CSS
+
+<iframe width="800" height="450" src="//embed.chartblocks.com/1.0/?c=5d1a0f363ba0f6d81845627b&t=2999c8ac5b1c983" frameBorder="0"></iframe>
+
+### Hello world weight vs real world weight
+
+When it comes to JS Web Frameworks, figures has to be discussed furthermore.
+
+Modern frameworks are deeply modular today. As much Angular which consider each modules as part of the framework or React which consider each as third parties. Yet you often need these modules for your app then considering only the core framework weight is a mistake.
+
+It can create an important difference between the weight of an hello world app using only the core module of a framework and a real world app. The extensive use of Tree Shaking recently amplified the problem. The risk is that some framework would communicate on cool numbers with a very simple example using nothing and where Tree Shaking removed literally everything when the true size of the framework has not changed.
+
+<iframe width="800" height="450" src="//embed.chartblocks.com/1.0/?c=5d1a1d2f3ba0f6d71e45627a&t=462bcc52236cd5f" frameBorder="0"></iframe>
+
+In these measures, I bundled app with the framework, a state management framework, a router and a Material Design component library (with an import all to prevent Tree Shaking). We see that the initial weight of a framework does not necessarily indicates the final result. Yet, it looks like the Vue figure is mainly due to Vuetify which contains really a lot of components.
```

Comment:
```suggestion
To perform these measures, I bundled the app with the framework, a state management framework, a router and a Material Design component library (with an import all to prevent Tree Shaking). We see that the initial weight of a framework does not necessarily indicates the final result. Yet, it looks like the Vue figure is mainly due to Vuetify which contains really a lot of components.
```